synapseclient/core/version_check.py

```diff
@@ -11,150 +11,274 @@
 
 """
 
-import importlib.resources
 import json
+import logging
 import re
 import sys
+from importlib.resources import files
+from typing import Any, Optional
 
-import requests
+import httpx
 
 import synapseclient
 
-_VERSION_URL = "https://raw.githubusercontent.com/Sage-Bionetworks/synapsePythonClient/master/synapseclient/synapsePythonClient"  # noqa
+_PYPI_JSON_URL = "https://pypi.org/pypi/synapseclient/json"
+_RELEASE_NOTES_URL = "https://python-docs.synapse.org/news/"
 
 
 def version_check(
-    current_version=None, version_url=_VERSION_URL, check_for_point_releases=False
-):
+    current_version: Optional[str] = None,
+    check_for_point_releases: bool = False,
+    use_local_metadata: bool = False,
+    logger: logging.Logger = None,
```

Comment:
Now that `logger` can be passed into this function, we'll need to pass the logger from where it's being called in the Synapse class

Reply:
Done